### PR TITLE
New gizmo structure and gizmo disabling menu

### PR DESCRIPTION
**Pull request by JFonS**

### CODE REVIEW

File: editor/plugins/spatial_editor_plugin.cpp [hunk 1]
```diff
@@ -5596,3 +5606,169 @@ SpatialEditorPlugin::SpatialEditorPlugin(EditorNode *p_node) {
...
+
+String GizmoPlugin::get_name() const {
+	return "The rest";
+}
+
+bool GizmoPlugin::can_be_hidden() const {
+	return true;
+}
+
+void GizmoPlugin::redraw(SpatialEditorGizmo *p_gizmo) {
```
Comment (reduz) [inline editor/plugins/spatial_editor_plugin.cpp]: I think the gizmo is a reference, so passing a reference should be fine, gizmo can create it on its own as:
Ref<SpatialEditorGizmo>(this) when calling the plugin.

File: editor/editor_node.cpp [hunk 2]
```diff
@@ -3068,7 +3068,7 @@ void EditorNode::register_editor_types() {
 	ClassDB::register_class<EditorSelection>();
 	ClassDB::register_class<EditorFileDialog>();
 	ClassDB::register_virtual_class<EditorSettings>();
-	ClassDB::register_class<EditorSpatialGizmo>();
+	ClassDB::register_class<SpatialEditorGizmo>();
```
Comment (groud) [inline editor/editor_node.cpp]: Why renaming this class ? All editor classes start with Editor, so it is easier to spot which classes are editor only.

Comment (JFonS) [reply editor/editor_node.cpp]: This PR merges SpatialEditorGizmo and EditorSpatialGizmo into a single class. I picked SpatialEditorGizmo as the name of the merged class but I don't mind changing it to the other.

Comment (groud) [reply editor/editor_node.cpp]: If it is an Editor only class, I think keeping EditorSpatialGizmo is better. But it is not that important ^^

**Pull request merged**